src/main/java/teammates/ui/automated/PrepareEmailWorkerAction.java

```diff
@@ -0,0 +1,163 @@
+        
+        String feedbackSessionName = getRequestParamValue(ParamsNames.EMAIL_FEEDBACK);
+        
+        String courseId = getRequestParamValue(ParamsNames.EMAIL_COURSE);
+        Assumption.assertNotNull(courseId);
+        
+        log.info("Prepare email worker activated for: " + HttpRequestHelper.printRequestParameters(request));
+        
+        List<EmailWrapper> emailsToBeSent = prepareEmailsToBeSent(emailType, feedbackSessionName, courseId);
+        
+        if (emailsToBeSent == null) {
+            log.severe("Emails to be sent is null for type: " + emailTypeString);
+            setErrorResponse();
+            return;
+        }
+        
+        try {
+            new EmailSender().sendEmails(emailsToBeSent);
+            doPostProcessingForSuccesfulSend(emailType, feedbackSessionName, courseId);
+        } catch (Exception e) {
+            try {
+                doPostProcessingForUnsuccesfulSend(emailType, courseId);
+            } catch (EntityDoesNotExistException ednee) {
+                log.severe("Unexpected error: " + TeammatesException.toStringWithStackTrace(ednee));
+            }
+            log.severe("Unexpected error: " + TeammatesException.toStringWithStackTrace(e));
+            setErrorResponse();
+        }
+    }
+    
+    private void doPostProcessingForSuccesfulSend(EmailType emailType, String feedbackSessionName, String courseId)
+            throws InvalidParametersException, EntityDoesNotExistException {
+        switch (emailType) {
+        case FEEDBACK_CLOSING:
+        case FEEDBACK_CLOSED:
+        case FEEDBACK_OPENING:
+        case FEEDBACK_PUBLISHED:
+        case FEEDBACK_UNPUBLISHED:
+            FeedbackSessionAttributes feedbackSession = fsLogic.getFeedbackSession(feedbackSessionName, courseId);
+            switch (emailType) {
+            case FEEDBACK_CLOSING:
+                feedbackSession.setSentClosingEmail(true);
+                break;
+            case FEEDBACK_CLOSED:
+                feedbackSession.setSentClosedEmail(true);
+                break;
+            case FEEDBACK_OPENING:
+                feedbackSession.setSentOpenEmail(true);
+                break;
+            case FEEDBACK_PUBLISHED:
+                feedbackSession.setSentPublishedEmail(true);
+                break;
+            case FEEDBACK_UNPUBLISHED:
+                feedbackSession.setSentPublishedEmail(false);
+                break;
+            default: // impossible to reach
+                break;
+            }
+            fsLogic.updateFeedbackSession(feedbackSession);
+            break;
+            
+        case PENDING_COMMENT_CLEARED:
+            frcLogic.updateFeedbackResponseCommentsSendingState(
+                    courseId, CommentSendingState.SENDING, CommentSendingState.SENT);
+            commentsLogic.updateCommentsSendingState(courseId, CommentSendingState.SENDING, CommentSendingState.SENT);
+            break;
+            
+        default:
```

Comment:
Actually, this `default` can never be reached. This method is called at line 63, but line 55 would already cause the method to terminate when the condition is the same as the condition that fulfills this `default`.
So if there's a place where I can do what I've described, it would be on the method `prepareEmailsToBeSent`.